src/utils/distributions.cpp

```diff
@@ -2,5 +2,15 @@
 
 int bayesmix::categorical_rng(Eigen::VectorXd probas, std::mt19937_64 rng,
                               int start /*= 0*/) {
-    return stan::math::categorical_rng(probas, rng) - (start+1);
+  return stan::math::categorical_rng(probas, rng) - (start+1);
+}
+
+double bayesmix::multi_normal_prec_lpdf(Eigen::VectorXd datum,
+                                        Eigen::VectorXd mean,
+                                        Eigen::MatrixXd prec_chol,
+                                        double prec_logdet) {
+  using stan::math::NEG_LOG_SQRT_TWO_PI;
+  double base = 0.5 * prec_logdet + NEG_LOG_SQRT_TWO_PI * datum.size();
+  double exp = 0.5 * (prec_chol * (datum - mu)).squaredNorm();
+  return base - exp;
```

Comment:
please check this formula, the multivariate normal density is

(2 * pi)^(- k / 2) * det(prec)^(0.5) * exp(- 1/2 (datum - mean)^T * prec * (datum - mean))

so that taking the log should be

-k / 2 * log(2 * pi) + 0.5 * det(prec) + 0.5 * (datum - mean)^T * prec * (datum - mean)

the last term is 'exp', while I think in 'base' you're missing a 0.5 somewhere

Reply:
Correct, thank you. This has now been fixed